### PR TITLE
Optionally display error messages if expansion fails with expandcmd()

### PR DESCRIPTION
**Pull request by yegappan**

### CODE REVIEW

**Comment by brammool** [comment]
ASAN reports a leak, perhaps something is not cleared on failure?

File: src/evalfunc.c [hunk 1]
```diff
@@ -4152,10 +4152,17 @@ f_expandcmd(typval_T *argvars, typval_T *rettv)
...
+    int		emsgoff = TRUE;
 
-    if (in_vim9script() && check_for_string_arg(argvars, 0) == FAIL)
+    if (in_vim9script()
+	    && (check_for_string_arg(argvars, 0) == FAIL
+		|| check_for_opt_dict_arg(argvars, 1) == FAIL))
 	return;
 
+    if (argvars[1].v_type == VAR_DICT
+	    && dict_has_key(argvars[1].vval.v_dict, "errmsg", -1))
```
Comment (LemonBoy) [inline src/evalfunc.c]: Shouldn't this be `dict_get_bool`? if `errmsg` is present but set to `v:false` then `emsgoff` should not be set to false.

**Comment by vim-ml** [comment]
Hi Bram,

On Sun, Apr 3, 2022 at 10:52 AM Bram Moolenaar ***@***.***>
wrote:

> ASAN reports a leak, perhaps something is not cleared on failure?
>
>
>
Yes. The new test exposed an existing memory leak. I have updated the PR
with a fix.

- Yegappan

**Comment by vim-ml** [comment]
Hi,

On Sun, Apr 3, 2022 at 11:14 AM LemonBoy ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In src/evalfunc.c
> <https://github.com/vim/vim/pull/10071#discussion_r841258614>:
>
> >  	return;
>
> +    if (argvars[1].v_type == VAR_DICT
> +	    && dict_has_key(argvars[1].vval.v_dict, "errmsg", -1))
>
> Shouldn't this be dict_get_bool? if errmsg is present but set to v:false
> then emsgoff should not be set to false.
>
>
>
Yes. I have updated the PR with this change.

- Yegappan

**Pull request closed without merging**